hub/util/exceptions.py

```diff
@@ -397,17 +390,26 @@ def __init__(self, tensors, output_keys):
 
 
 class InvalidOutputDatasetError(TransformError):
-    def __init__(self):
-        super().__init__(
-            "One or more tensors of the ds_out have different lengths. Transform only supports ds_out having same number of samples for each tensor (This includes empty datasets that have 0 samples per tensor)."
-        )
+    def __init__(
+        self, message="The output Dataset to transform should not be in read mode."
+    ):
+        super().__init__(message)
 
 
-class MemoryDatasetNotSupportedError(TransformError):
-    def __init__(self, scheduler):
-        super().__init__(
-            f"Transforms with ds_out having base storage as MemoryProvider are only supported in threaded mode. Current mode is {scheduler}."
-        )
+class TransformComposeEmptyListError(TransformError):
+    def __init__(
+        self,
+        message="The list passed to hub.compose is empty. Ensure that hub.compose is always called with a list of 1 or more functions.",
+    ):
+        super().__init__(message)
+
+
+class TransformComposeIncompatibleFunction(TransformError):
+    def __init__(
+        self,
+        message="One or more elements passed to hub.compose are incompatible with it. Ensure that functions are all decorated with hub.compute decorator and instead of passing my_fn, pass my_fn() in the list.",
+    ):
```

Comment:
```suggestion
        message="One or more elements passed to hub.compose are incompatible with it. Ensure that functions are all decorated with hub.compute decorator and instead of passing my_fn, pass my_fn() in the list.",
        message=f"hub.compose received an incompatible element at index {index}. Expected "{expected_type}" but got "{actual_type}".",
    ):
```

Reply:
May need to change the wording and stuff, but would be nice to know exactly what index failed and why

Reply:
This exception is about hub.compose failing, not transform. hub.compose can fail if you pass some wrong function to it.